resources/UserValidation.ts

```diff
@@ -0,0 +1,58 @@
+import { User } from './interfaces'
+import { UserData } from "./UserData";
+
+export class UserValidation {
+
+    private userData: UserData;
+
+    constructor(userData: UserData) {
+        this.userData = userData;
+    }
+
+    private isEmptyField(email: string, password: string): boolean {
+        return email == "" || password == "";
+    }
+    
+    private isInvalidEmail(email: string): boolean {
+        const expression: RegExp = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+.[A-Z]{2,}$/i; 
+        return !expression.test(email);
+    }
+
+    private isInvalidPassword(password: string): boolean {
+        const expression: RegExp = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/;
+        return !expression.test(password);
+    }
+
+    private isExistingUser(email: string): boolean {
+        const currentUser: User = this.userData.getUserByEmail(email);
+        return currentUser.email !== "" && currentUser.password !== "";
+    }
+
+    public registerValidation(email: string, password: string): string {
```

Comment:
Think about how you can improve the outcome of this thinking that multiple errors can occur.